test/js-native-api/test_finalizer/test.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+// Flags: --expose-gc
+
+const common = require('../../common');
+const test_finalizer = require(`./build/${common.buildType}/test_finalizer`);
+const assert = require('assert');
+
+(() => {
+  const obj = {};
+  test_finalizer.addFinalizer(obj);
+})();
+
+for (let i = 0; i < 10; ++i) {
+  global.gc();
+  if (test_finalizer.getFinalizerCallCount() === 1) {
+    break;
+  }
+}
+
+assert.strictEqual(test_finalizer.getFinalizerCallCount(), 1);
+
+async function runAsyncTests() {
+  let js_is_called = false;
```

Comment:
Maybe add a comment here explaining that we cannot use `common.mustCall()` because we want to make sure that the function gets called during gc, meaning that it will have to have been called by the time we make the assertion. It's not sufficient that it get called anytime before the process exits.

Reply:
I have added a comment.